src/EFCore.MySql/Storage/Internal/MySqlJsonTypeMapping.cs

```diff
@@ -20,34 +22,75 @@ namespace EFCore.MySql.Storage.Internal
     /// </summary>
     public class MySqlJsonTypeMapping : RelationalTypeMapping
     {
+        private static readonly Dictionary<Type, ValueConverter> JsonConverters = new Dictionary<Type, ValueConverter>();
+        private static readonly Dictionary<Type, ValueComparer> JsonComparers = new Dictionary<Type, ValueComparer>();
+
         /// <summary>
-        ///     Initializes a new instance of the <see cref="MySqlJsonTypeMapping" /> class.
+        ///     This API supports the Entity Framework Core infrastructure and is not intended to be used
+        ///     directly from your code. This API may change or be removed in future releases.
         /// </summary>
-        /// <param name="storeType"> The name of the database type. </param>
-        /// <param name="dbType"> The <see cref="System.Data.DbType" /> to be used. </param>
-        /// <param name="unicode"> A value indicating whether the type should handle Unicode data or not. </param>
-        /// <param name="size"> The size of data the property is configured to store, or null if no size is configured. </param>
         public MySqlJsonTypeMapping(
-            [NotNull] string storeType,
-            [CanBeNull] DbType? dbType = null,
-            bool unicode = false,
-            int? size = null)
-            : base(storeType, typeof(JsonObject<>), dbType, unicode, size)
+            Type clrType,
+            string storeType = null,
+            bool? unicode = null)
+            : this(
+                new RelationalTypeMappingParameters(
+                    new CoreTypeMappingParameters(clrType, GetConverter(clrType), GetComprarer(clrType)),
+                    storeType ?? "json",
+                    StoreTypePostfix.None,
+                    System.Data.DbType.String,
+                    unicode ?? false))
+        {
+        }
+
+        private static ValueConverter GetConverter(Type jsonType)
```

Comment:
Do these need to be thread-safe?

Reply:
At worst it would just create the converter twice.

Reply:
But TryGetValue and insertion into the Dictionary is not thread-safe, so if two threads try to do this at the same time it would not be safe--use a ConcurrentDictionary instead?